ext/java/org/msgpack/jruby/Decoder.java

```diff
@@ -118,7 +140,20 @@ private IRubyObject consumeHash(int size) {
   private IRubyObject consumeExtension(int size) {
     int type = buffer.get();
     byte[] payload = readBytes(size);
-    return ExtensionValue.newExtensionValue(runtime, type, payload);
+
+    if (this.registry != null) {
+      IRubyObject proc = this.registry.lookup(type);
```

Comment:
You don't need to prefix all instance variable accesses with `this.`. It's not that it matters if you do, just that it breaks the style of the rest of the Java code.